lib/fetch/index.js

```diff
@@ -122,130 +122,178 @@ class Fetch extends EE {
+      // 1. If locallyAborted is true, terminate these substeps.
+      if (locallyAborted) {
+        return Promise.resolve()
+      }
 
-    // 2. If response’s aborted flag is set, then:
-    if (response.aborted) {
-      // 1. Let deserializedError be the result of deserialize a serialized
-      //    abort reason given controller’s serialized abort reason and
-      //    relevantRealm.
+      try {
+        // 2. If response’s aborted flag is set, then:
+        if (response.aborted) {
+          // 1. Let deserializedError be the result of deserialize a serialized
+          //    abort reason given controller’s serialized abort reason and
+          //    relevantRealm.
 
-      // 2. Abort the fetch() call with p, request, responseObject, and
-      //    deserializedError.
+          // 2. Abort the fetch() call with p, request, responseObject, and
+          //    deserializedError.
 
-      abortFetch(p, request, responseObject, controller.serializedAbortReason)
-      return Promise.resolve()
-    }
+          abortFetch(p, request, responseObject, controller.serializedAbortReason, context)
+          return Promise.resolve()
+        }
 
-    // 3. If response is a network error, then reject p with a TypeError
-    // and terminate these substeps.
-    if (response.type === 'error') {
-      p.reject(new TypeError('fetch failed', { cause: response.error }))
-      return Promise.resolve()
-    }
+        // 3. If response is a network error, then reject p with a TypeError
+        // and terminate these substeps.
+        if (response.type === 'error') {
+          context.error = response.error
+          channels.error.publish(context)
+          delete context.error
+          p.reject(new TypeError('fetch failed', { cause: response.error }))
+          return Promise.resolve()
+        }
 
-    // 4. Set responseObject to the result of creating a Response object,
-    // given response, "immutable", and relevantRealm.
-    responseObject = fromInnerResponse(response, 'immutable', relevantRealm)
+        // 4. Set responseObject to the result of creating a Response object,
+        // given response, "immutable", and relevantRealm.
+        responseObject = fromInnerResponse(response, 'immutable', relevantRealm)
 
-    // 5. Resolve p with responseObject.
-    p.resolve(responseObject)
-  }
+        context.result = responseObject[kState]
 
-  controller = fetching({
-    request,
-    processResponseEndOfBody: handleFetchDone,
-    processResponse,
-    dispatcher: init?.dispatcher ?? getGlobalDispatcher() // undici
-  })
+      // 5. Resolve p with responseObject.
+      } finally {
+        callAsyncStartEnd()
+        p.resolve(responseObject)
+      }
+    }
 
+    controller = fetching({
+      request,
+      processResponseEndOfBody: handleFetchDone,
+      processResponse,
+      dispatcher: init?.dispatcher ?? getGlobalDispatcher() // undici
+    })
+  } finally {
+    channels.end.publish(context)
```

Comment:
Same as per subscribers check

Reply:
updated all events to check for subscribers